ovos_dinkum_listener/voice_loop/voice_loop.py

```diff
@@ -241,28 +250,54 @@ def reset_state(self):
         LOG.debug(f"state={self.state}|hotwords.state={self.hotwords.state}")
 
     def go_to_sleep(self):
+        """
+        Set the Listening State to "Sleeping" until woken
+        """
         self.state = ListeningState.SLEEPING
         LOG.info("sleeping")
 
     def wakeup(self):
+        """
+        Reset the Listening State from "Sleeping" to the default for the current
+        Listening Mode.
+        """
         self.reset_state()
         LOG.info("wakeup")
 
-    def start_recording(self, filename=None):
+    def start_recording(self, filename: Optional[str] = None):
+        """
+        Set the listening state to RECORDING and specify a file to record to
+        @param filename: filename to record mic input to
+        """
         self.recording_filename = filename or str(time.time())
         LOG.debug(f"Recording to {self.recording_filename}")
         self.state = ListeningState.RECORDING
 
     def stop_recording(self):
+        """
+        Stop recording, pass audio and metadata (recording filename) to the
+        `recording_audio_callback` method and reset the Listening State
+        """
         #  finished recording
         if self.recording_audio_callback is not None:
             metadata = {"recording_name": self.recording_filename}
-            metadata = self.recording_audio_callback(self.stt_audio_bytes,
-                                                     metadata) or metadata
+            self.recording_audio_callback(self.stt_audio_bytes, metadata)
         LOG.debug("Finished recording")
         self.reset_state()
 
-    def _in_recording(self, chunk):
+    def _in_recording(self, chunk: bytes):
+        """
+        Handle a chunk of audio while in the `RECORDING` state
+        (i.e. after wake word detection). Check for stop words in all cases and
+        pass audio to any loaded audio transformers.
+
+        If a "stop" hotword is detected, the appropriate method is called.
+
+        If no "stop" hotword is detected, audio is evaluated by VAD and any
+        speech frames are passed to audio transformers.
```

Comment:
no vad is used here

Reply:
that is just tagging the chunk object for further downstream reuse if needed, but all frames are passed to audio transformers regardless of that config...